exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/StatusResources.java

```diff
@@ -50,75 +53,124 @@
 public class StatusResources {
   static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(StatusResources.class);
 
+  public static final String PATH_STATUS_JSON = "/status.json";
+  public static final String PATH_STATUS = "/status";
+  public static final String PATH_OPTIONS_JSON = "/options.json";
+  public static final String PATH_INTERNAL_OPTIONS_JSON = "/internal_options.json";
+  public static final String PATH_OPTIONS = "/options";
+  public static final String PATH_INTERNAL_OPTIONS = "/internal_options";
+
   @Inject UserAuthEnabled authEnabled;
   @Inject WorkManager work;
   @Inject SecurityContext sc;
 
   @GET
-  @Path("/status.json")
+  @Path(StatusResources.PATH_STATUS_JSON)
   @Produces(MediaType.APPLICATION_JSON)
   public Pair<String, String> getStatusJSON() {
     return new ImmutablePair<>("status", "Running!");
   }
 
   @GET
-  @Path("/status")
+  @Path(StatusResources.PATH_STATUS)
   @Produces(MediaType.TEXT_HTML)
   public Viewable getStatus() {
     return ViewableWithPermissions.create(authEnabled.get(), "/rest/status.ftl", sc, getStatusJSON());
   }
 
-  @GET
-  @Path("/options.json")
-  @RolesAllowed(DrillUserPrincipal.AUTHENTICATED_ROLE)
-  @Produces(MediaType.APPLICATION_JSON)
-  public List<OptionWrapper> getSystemOptionsJSON() {
+  private List<OptionWrapper> getSystemOptionsJSONHelper(boolean internal)
+  {
     List<OptionWrapper> options = new LinkedList<>();
-    for (OptionValue option : work.getContext().getOptionManager()) {
+    OptionManager optionManager = work.getContext().getOptionManager();
+    OptionList optionList = internal ? optionManager.getInternalOptionList(): optionManager.getExternalOptionList();
+
+    for (OptionValue option : optionList) {
       options.add(new OptionWrapper(option.name, option.getValue(), option.type, option.kind));
     }
+
```

Comment:
FWIW, DRILL-5716 introduces name sorting to this page. (I can never find anything in the original unsorted list...)